utilities/openshift_resources/cluster_scoped_resource.py

```
@@ -0,0 +1,300 @@
+        }
+        if self._label:
+            resource["metadata"]["labels"] = self._label
+        if self._annotations:
+            resource["metadata"]["annotations"] = self._annotations
+        return resource
+
+    def _write(self, resource_dict: dict[str, Any], verb: str) -> Self:
+        with tempfile.NamedTemporaryFile(mode="w", suffix=".json", delete=True) as tmp:
+            json.dump(resource_dict, tmp)
+            tmp.flush()
+            run_oc(verb, "-f", tmp.name, "-o", "json")  # noqa: FCN001 - run_oc uses *args
```

Ruff (0.16.0)

[warning] Invalid rule code in `# noqa`: FCN001

Add non-Ruff rule codes to the `lint.external` configuration option

(RUF102)